include/rpc/detail/make_unique.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#ifndef MAKE_UNIQUE_H_FOOBAR
+#define MAKE_UNIQUE_H_FOOBAR
+
+#include <memory>
+
+namespace rpc {
+namespace detail {
+
+using std::make_unique;
```

Comment:
I think it's enough to create a wrapper that has a conditional implementation (one forwarding the arguments to std::make_unique, when available, other being the hand-rolled make_unique).